scripts/bumpversion.mjs

```diff
@@ -0,0 +1,189 @@
+    {
+      name: "tag",
+      type: "list",
+      message: `Are you sure you want to release to npm@${tag}`,
+      choices: ["yes", "no", "choose different tag"],
+    },
+  ]);
+
+  if (proceed === "choose different tag") {
+    return chooseTag();
+  }
+
+  if (proceed === "no") {
+    return;
+  }
+
+  return tag;
+}
+
+async function main() {
+  if (!process.env.PERCY_TOKEN) {
+    console.log(chalk.red("No PERCY_TOKEN found in the environment"));
+    return;
+  }
+
+  const branch = await runSilently(() => $`git branch --show-current`);
+
+  if (branch !== "master") {
+    const { proceed } = await inquirer.prompt([
+      {
+        name: "proceed",
+        type: "list",
+        message: `You are currently on branch ${branch}, do you wish to proceed?`,
+        choices: ["switch to latest master", "yes", "no"],
+      },
+    ]);
+
+    if (proceed === "no") {
+      console.log(chalk.red("aborting the release"));
+      return;
+    }
+
+    if (proceed === "switch to latest master") {
+      await $`git checkout master`;
+      await $`git pull`;
+    }
+  }
+
+  let { version } = await require("../package.json");
+  console.log("current version: ", chalk.blue(version));
+
+  const versions = [
+    "prerelease",
+    "patch",
+    "premajor",
+    "preminor",
+    "prepatch",
+    "major",
+    "minor",
+    "pre",
+  ];
+
+  const { release } = await inquirer.prompt([
+    {
+      name: "release",
+      type: "list",
+      message: "Choose the type of release",
+      choices: versions,
+    },
+  ]);
+
+  const newVersion = semver.inc(version, release);
+
+  const { proceed } = await inquirer.prompt([
+    {
+      name: "proceed",
+      type: "list",
+      message: `The new version will be ${newVersion}, do you wish to proceed?`,
+      choices: ["yes", "no"],
+    },
+  ]);
+
+  if (proceed === "no") {
+    console.log(chalk.red("aborting the release"));
+    return;
+  }
+  const content = await fs.readFile("./package.json");
+  const replaced = content.toString().replace(version, newVersion);
+  await fs.writeFile("./package.json", replaced);
+  console.log(chalk.green("bumped version in package"));
+
+  await runWithSpinner(() => $`yarn`, "installing dependencies");
+  console.log(chalk.green("installed dependencies"));
+  await runWithSpinner(() => $`yarn build`, "building basis");
+  console.log(chalk.green("build complete"));
+  const percyOutput = await runWithSpinner(
+    () => $`yarn percy`,
+    "running percy build"
+  );
+  console.log(chalk.green("percy build complete"));
+  const buildUrl = percyOutput.match(/(https:\/\/percy.io\S*)/g)[0];
+  await sleep(1000);
+
+  const { passes } = await inquirer.prompt([
+    {
+      name: "passes",
+      type: "list",
+      message: `Confirm that the following percy build passes: ${buildUrl}`,
+      choices: ["yes", "no"],
+    },
+  ]);
+
+  if (passes === "no") {
+    console.log(chalk.red("aborting the release"));
+    return;
```

Comment:
Wouldn't it be `process.exit(1)`

Reply:
I was fine with returning without really throwing exception but I could do that